source/frontend/editors/RarcEditor.hpp

```diff
@@ -0,0 +1,255 @@
+#pragma once
+
+#include <frontend/IEditor.hpp>
+#include <frontend/legacy_editor/StudioWindow.hpp>
+#include <frontend/widgets/PropertyEditorWidget.hpp>
+#include <fstream>
+#include <librii/rarc/RARC.hpp>
+#include <librii/szs/SZS.hpp>
+#include <rsl/FsDialog.hpp>
+
+namespace riistudio::frontend {
+
+// Implements a single tab for now
+class RarcEditorTabSheet : private PropertyEditorWidget {
+public:
+  void Draw(std::function<void(void)> draw) {
+    // No concurrent access
+    assert(m_drawTab == nullptr);
+    m_drawTab = draw;
+    DrawTabWidget(false);
+    Tabs();
+    m_drawTab = nullptr;
+  }
+
+private:
+  std::vector<std::string> TabTitles() override { return {"RARC"}; }
+  bool Tab(int index) override {
+    m_drawTab();
+    return true;
+  }
+
+  std::function<void(void)> m_drawTab = nullptr;
+};
+
+enum class TriState {
+	ST_FALSE,
+	ST_TRUE,
+	ST_INDETERMINATE
+};
+
+enum class ModalState {
+	M_CLOSED,
+	M_OPENING,
+	M_OPEN
+};
+
+class RarcEditor;
+
+class RarcEditorPropertyGrid {
+  friend class RarcEditor;
+
+public:
+  RarcEditorPropertyGrid() = default;
+  void Draw(librii::RARC::ResourceArchive& arc, RarcEditor* editor);
+
+private:
+  void DrawTable(librii::RARC::ResourceArchive& arc, RarcEditor* editor);
+  void DrawFlagsColumn(librii::RARC::ResourceArchive::Node& node,
+                       RarcEditor* editor);
+  void DrawSizeColumn(librii::RARC::ResourceArchive::Node& node,
+                      RarcEditor* editor);
+  void DrawContextMenu(librii::RARC::ResourceArchive::Node& node,
+                       RarcEditor* editor);
+  void DrawNameModal(RarcEditor* editor);
+  void DrawOperationModal(RarcEditor* editor);
+
+  ImGuiTextFilter m_model_filter;
+  std::optional<librii::RARC::ResourceArchive::Node> m_focused_node;
+  std::string m_name_input;
+  std::string m_original_name;
+  ModalState m_context_modal_state;
+  ModalState m_operation_modal_state;
+  TriState m_operation_state;
+  bool m_should_calc_changes;
+};
+
+class RarcEditor : public frontend::StudioWindow, public IEditor {
+public:
+  RarcEditor()
+      : StudioWindow("RARC Editor: <unknown>", DockSetting::None), m_grid(),
+        m_rarc(), m_dir_size_cache(), m_changes_made(), m_node_to_rename(),
+        m_node_new_name(), m_files_to_insert(), m_folder_to_insert(),
+        m_folder_to_create(), m_node_to_extract(), m_extract_path(),
+        m_node_to_replace(), m_replace_path() {
+    // setWindowFlag(ImGuiWindowFlags_MenuBar);
+  }
```

Comment:
Modern C++ style would be to initialize these fields directly in their declaration, e.g.:
`bool m_changes_made = false;` or `bool m_changes_made{false};` or `bool m_changes_made{};`